### PR TITLE
1231 Solana client

### PR DESCRIPTION
**Pull request by 0xp3gasus**

### CODE REVIEW

File: examples/solana/.codesandbox/tasks.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    // These tasks will run in order when initializing your CodeSandbox project.
+    "setupTasks": [
+      {
+        "name": "Install Dependencies",
+        "command": "yarn install"
+      }
+    ]
+}
```
Comment (juanfri-web3) [inline examples/solana/.codesandbox/tasks.json:9]:
```suggestion
}

```

File: examples/solana/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "name": "Devcontainer",
+    "image": "ghcr.io/codesandbox/devcontainers/typescript-node:latest"
+}
```
Comment (juanfri-web3) [inline examples/solana/.devcontainer/devcontainer.json:4]:
```suggestion
}

```

File: examples/solana/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "module":"commonjs",
+        "target": "es5",
+        "noEmitOnError": true,
+        "resolveJsonModule": true,
+        "esModuleInterop": true,
+        "lib": ["es6", "dom", "es2016", "es2017"]
+    }
+}
```
Comment (juanfri-web3) [inline examples/solana/tsconfig.json:10]:
```suggestion
}

```

File: packages/xchain-solana/package.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    "@xchainjs/xchain-crypto": "workspace:*",
+    "@xchainjs/xchain-util": "workspace:*",
+    "bs58": "6.0.0",
+    "micro-ed25519-hdkey": "0.1.2"
+  },
+  "publishConfig": {
+    "access": "public",
+    "directory": "release/package"
+  }
+}
```
Comment (juanfri-web3) [inline packages/xchain-solana/package.json:51]:
```suggestion
}

```

File: packages/xchain-solana/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "../../tsconfig.json",
+  "include": [
+    "src/**/*"
+  ]
+}
```
Comment (juanfri-web3) [inline packages/xchain-solana/tsconfig.json:6]:
```suggestion
}

```

File: packages/xchain-solana/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Cluster } from '@solana/web3.js'
+import { Network } from '@xchainjs/xchain-client'
+
+export const getSolanaNetwork = (network: Network): Cluster => {
```
Comment (juanfri-web3) [inline packages/xchain-solana/src/utils.ts:4]: I did't test it, but WDYT about using a literal type here?
Something like this:
```typescript
export const getSolanaNetwork = (network: Network): Cluster => {
  const networkMap: { [key in Network]: Cluster } = {
    [Network.Mainnet]: 'mainnet-beta',
    [Network.Stagenet]: 'mainnet-beta',
    [Network.Testnet]: 'testnet',
  };

  return networkMap[network];
}
```

Comment (0xp3gasus) [reply packages/xchain-solana/src/utils.ts:4]: Good approach 😎

**Review by juanfri-web3 — APPROVED** [review]
Great!! 🚀

**Pull request merged**